pynestml/src/main/grammars/org/PyNESTML.g4

```diff
@@ -0,0 +1,329 @@
+  * Procedural-Language
+  *********************************************************************************************************************/
+
+  block : ( stmt | NEWLINE )*;
+
+  stmt : small_Stmt | compound_Stmt;
+
+  compound_Stmt : if_Stmt
+                | for_Stmt
+                | while_Stmt;
+
+  small_Stmt : assignment
+             | functionCall
+             | declaration
+             | returnStmt;
+
+  assignment : lhsVariable=variable
+    (directAssignment='='       |
+     compoundSum='+='     |
+     compoundMinus='-='   |
+     compoundProduct='*=' |
+     compoundQuotient='/=') expression;
+
+
+  /** ASTDeclaration A variable declaration. It can be a simple declaration defining one or multiple variables:
+   'a,b,c real = 0'. Or an function declaration 'function a = b + c'.
+    @attribute hide is true iff. declaration is not trackable.
+    @attribute function is true iff. declaration is an function.
+    @attribute vars          List with variables
+    @attribute Datatype      Obligatory data type, e.g. 'real' or 'mV/s'
+    @attribute sizeParameter An optional array parameter. E.g. 'tau_syn ms[n_receptros]'
+    @attribute expr An optional initial expression, e.g. 'a real = 10+10'
+    @attribute invariants List with optional invariants.
+   */
+  declaration :
+    isRecordable='recordable'? isFunction='function'?
+    variable (',' variable)*
+    datatype
+    ('[' sizeParameter=NAME ']')?
+    ( '=' rhs = expression)? SL_COMMENT?
+    ('[[' invariant=expression ']]')?;
+
+  /** ATReturnStmt Models the return statement in a function.
+
+    @attribute minus An optional sing
+    @attribute definingVariable Name of the variable
+   */
+  returnStmt : 'return' expression?;
+
+  if_Stmt : if_Clause
+            elif_Clause*
+            (else_Clause)?
+            BLOCK_CLOSE;
+
+  if_Clause : 'if' expression BLOCK_OPEN block;
+
+  elif_Clause : 'elif' expression BLOCK_OPEN block;
+
+  else_Clause : 'else' BLOCK_OPEN block;
+
+  for_Stmt : 'for' var=NAME 'in' vrom=expression '...' to=expression 'step' step=signedNumericLiteral BLOCK_OPEN block BLOCK_CLOSE;
+
+  while_Stmt : 'while' expression BLOCK_OPEN block BLOCK_CLOSE;
+
+  signedNumericLiteral : (negative='-') NUMERIC_LITERAL;
+
+  /*********************************************************************************************************************
+  * Nestml-Language
+  *********************************************************************************************************************/
+  /** ASTNeuron represents neuron.
+    @attribute Name    The name of the neuron
+    @attribute Body    The body of the neuron, e.g. internal, state, parameter...
+  */
+  neuron : 'neuron' NAME body;
+
+  /** ASTBody The body of the neuron, e.g. internal, state, parameter...
+  */
+  body : BLOCK_OPEN
+           (NEWLINE | var_Block | dynamics | equations | inputBuffer | outputBuffer | function)*
+         BLOCK_CLOSE;
+
```

Comment:
I would use different names. E.g.:
`(NEWLINE | BlockWithVariables | UpdateBlock | EquationsBlock | InputBlock | OutputBlock | Function)*`

Reply:
This part has been reworked according to your feedback.